containers/podman-desktop
# feat: check connectivity with health check

### What does this PR do?

check connectivity with health check, in the context of experimental kubernetes states,
and saves the state in a `ContextsConnectivityRegistry`.

The next step is to dispatch the state to the frontend.

(The first commit of the PR is a copy of the one in https://github.com/podman-desktop/podman-desktop/pull/10074)

### Screenshot / video of UI

backend only

### What issues does this PR fix or reference?

Part of #9938 

### How to test this PR?

To test the new version, set this preference in ~/.local/share/containers/podman-desktop/configuration/settings.json:
```
  "kubernetes.statesExperimental": true
```

You can have some cluster behind a VPN to provoke some timeout and check by editing the kubeconfig file several times quickly (by changing the current context for example), that previous checks are aborted and only a response from the last (un-aborted) check is received.

Some logs can be added to the `ContextsConnectivityRegistry` to display the state each time it is changed
